map_claims.go

```diff
@@ -3,18 +3,44 @@ package jwt
 import (
 	"encoding/json"
 	"errors"
-	// "fmt"
+	//"fmt"
 )
 
 // Claims type that uses the map[string]interface{} for JSON decoding
 // This is the default claims type if you don't supply one
 type MapClaims map[string]interface{}
 
 // Compares the aud claim against cmp.
+// If the aud claim is a string, this method will return true if the value matches exactly.
+// If the aud claim is a slice of strings, this method will return true if the value exactly matches any of the items in the slice.
 // If required is false, this method will return true if the value matches or is unset
 func (m MapClaims) VerifyAudience(cmp string, req bool) bool {
-	aud, _ := m["aud"].(string)
-	return verifyAud(aud, cmp, req)
+	aud := m["aud"]
+
+	switch aud.(type) {
+	default:
```

Comment:
It works first, I saw this first at https://golang.org/doc/effective_go.html#type_switch . I agree it probably reads better last though.